packages/client/src/errors.ts

```diff
@@ -87,6 +90,36 @@ export function isGrpcServiceError(err: unknown): err is GrpcServiceError {
   );
 }
 
+/**
+ * Returns true if the provided error or its cause is a {@link GrpcServiceError} with code DEADLINE_EXCEEDED.
+ *
+ * @see {@link Connection.withDeadline}
+ */
+export function isGrpcDeadlineError(err: unknown): err is Error {
+  while (isError(err)) {
+    if (isGrpcServiceError(err) && (err as GrpcServiceError).code === status.DEADLINE_EXCEEDED) {
+      return true;
+    }
+    err = (err as any).cause;
+  }
+  return false;
+}
+
+/**
+ * Returns true if the provided error or its cause is a {@link GrpcServiceError} with code CANCELLED.
+ *
+ * @see {@link Connection.withAbortSignal}
+ */
+export function isGrpcCancelledError(err: unknown): err is Error {
+  while (isError(err)) {
+    if (isGrpcServiceError(err) && (err as GrpcServiceError).code === status.CANCELLED) {
+      return true;
+    }
+    err = (err as any).cause;
+  }
+  return false;
+}
+
```

Comment:
The logic is exactly the same for both, it just changes the constant, do you think is worthwhile to factor out that logic in a separate internal method that takes the constant as argument?

Reply:
Yeah… But not yet.

The length and complexity of the duplicated portion is very low, whereas the shared internal function we'd introduce to dedup this (something like `hasGrpcErrorAncestor(code)`? or maybe `hasErrorAncestor(predicateFn)`?) would add an unnatural abstraction with very limited potential of reuse afterward. I don't think that's a win at this point. Let's delay.